status/status.go

```diff
@@ -9,16 +10,92 @@ import (
 	"net"
 	"net/http"
 	"os"
+	"strings"
 	"sync/atomic"
 	"time"
 
+	"github.com/flynn/flynn/controller/utils"
 	"github.com/flynn/flynn/discoverd/cache"
 	"github.com/flynn/flynn/discoverd/client"
 	"github.com/flynn/flynn/pkg/status"
 )
 
 func main() {
-	log.Fatal(http.ListenAndServe(":"+os.Getenv("PORT"), status.Handler(GetStatus)))
+	handler := newStatusHandler(status.Handler(GetStatus), os.Getenv("AUTH_KEY"))
+	log.Fatal(http.ListenAndServe(":"+os.Getenv("PORT"), handler))
+}
+
+type statusHandler struct {
+	h         status.Handler
+	k         string
+	whitelist []*net.IPNet
+}
+
+func newStatusHandler(h status.Handler, k string) *statusHandler {
+	n := func(ip net.IP, n *net.IPNet, err error) *net.IPNet {
+		if err != nil {
+			panic(err)
+		}
+		return n
+	}
+	whitelist := []*net.IPNet{
+		n(net.ParseCIDR("10.0.0.0/8")),      // 10.0.0.0 - 10.255.255.255
+		n(net.ParseCIDR("172.16.0.0/12")),   // 172.16.0.0 - 172.31.255.255
+		n(net.ParseCIDR("192.168.0.0/16")),  // 192.168.0.0 - 192.168.255.255
+		n(net.ParseCIDR("127.0.0.0/8")),     // 127.0.0.0 - 127.255.255.255
+		n(net.ParseCIDR("fd00::/8")),        // fdxx:xxxx:xxxx...
+		n(net.ParseCIDR("::1/128")),         // ::1
+		n(net.ParseCIDR("192.0.2.0/24")),    // TEST-NET (for containers)
+		n(net.ParseCIDR("198.51.100.0/24")), // TEST-NET (for containers)
+		n(net.ParseCIDR("203.0.113.0/24")),  // TEST-NET (for containers)
+		n(net.ParseCIDR("100.64.0.0/10")),   // for containers
+	}
```

Comment:
It is conventional to prefix functions which are expected to always work with `must` (e.g. like [regexp.MustCompile](https://golang.org/pkg/regexp/#MustCompile)), so I would probably make this:

```go
whitelist := []*net.IPNet{
	mustParseCIDR("10.0.0.0/8"),
	mustParseCIDR("172.16.0.0/12"),
	...
}
```